modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -254,11 +306,12 @@ public function getBitcoinAmount($fiat_amount, $currency) {
 			if($margin > 0){
 				$price = $price * 100/(100+$margin);
 			}
+			return intval(1.0e8 * $fiat_amount/$price);
 		} catch (\Exception $e) {
 			echo "Error getting price from Blockonomics! {$e->getMessage()}";
 		}
 
-		return intval(1.0e8 * $fiat_amount/$price);
+		return;
```

Comment:
returning empty is a bad design. Best to let the exception propagate to parent or return an invalid value that the parent will check

Reply:
Moved in a4a5005